src/taipy/core/_entity/_labelled.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+from typing import Optional
+
+
+class _Labelled:
+    __LABEL_SEPARATOR = ">"
+
+    def get_label(self) -> str:
+        if explicit_label := self._get_explicit_label():
+            return explicit_label
+        return self._generate_label()
+
+    def get_simple_label(self) -> str:
+        if explicit_label := self._get_explicit_label():
+            return explicit_label
+        return self._generate_label(True)
+
+    def _generate_label(self, simple=False) -> str:
+        ls = []
+        if not simple:
+            if owner_id := self._get_owner_id():
+                if self.id != owner_id:  # type: ignore
+                    from ... import core
+
+                    owner = core.get(owner_id)
+                    ls.append(owner.get_label())
+        ls.append(self._generate_entity_label())
+        return self.__LABEL_SEPARATOR.join(ls)
+
+    def _get_explicit_label(self) -> Optional[str]:
+        try:
+            return self._properties.get("label")  # type: ignore
```

Comment:
I don't see where `self._properties` is initialized. Am I missing something?

Reply:
It is initialized in each child class (Scenario, DataNode, ...) It s because it is not initialized in the _labeled class, I encapsulated the code in a try-except block.